packages/upswyng-server/src/components/ResourceSearch.svelte

```diff
@@ -35,39 +43,37 @@
         .catch(e => (errorMessage = e.message))
         .finally(() => (isLoading = false));
     } else {
+      hasQuery = false;
       results = [];
     }
   }
 </script>
 
-<div class="field">
-  <div class="control has-icons-left" class:is-loading={isLoading}>
-    <label for="search" class="is-sr-only">Search for a provider</label>
-    <input
-      class="input is-rounded"
-      id="search"
-      name="search"
-      type="text"
-      placeholder="Search for a provider..."
-      bind:value={query} />
-    <span class="icon is-left">
-      <i class="fas fa-search" />
-    </span>
-  </div>
-  {#if query.length > 3 && !isLoading && !results.length}
-    <p class="help">No providers match your search</p>
-  {/if}
-  {#if errorMessage}
-    <p class="help is-danger">{errorMessage}</p>
-  {/if}
-</div>
-
-<div class="content">
-  {#each results as resource}
-    <ResourceSearchResult
-      {action}
-      name={resource.name}
-      description={resource.description}
-      on:resourceClick={() => dispatch('resourceClick', resource.objectID)} />
-  {/each}
+<div class="field dropdown is-block">
+  <label for="search" class="is-sr-only" id="search-label">
+    Search for a provider
+  </label>
+  <Autocomplete
+    className={isLoading ? 'is-flex is-loading' : 'is-flex'}
+    bind:value={query}
+    id="search"
+    placeholder="Search for a provider...">
+    <i class="fas fa-search" slot="input-left-icon" />
+    <div slot="help">
+      {#if hasQuery && !isLoading && !results.length}
+        <p class="help is-info">No providers match your search</p>
+      {/if}
+      {#if hasQuery && errorMessage}
+        <p class="help is-error">{errorMessage}</p>
+      {/if}
+    </div>
+    {#if hasQuery}
+      {#each results as resourceResult, i}
```

Comment:
probs should key this by the ID of the result.. I think as written is the default behavior but doesn't give you any benefits if the results change by the results length doesn't

Reply:
👍 [updated](https://github.com/CodeForBoulder/upswyng/pull/332/commits/391fc61f51613ecede33f4b95de61b35982ec4c8)